### PR TITLE
Feature/df 1827 from release df1505 ch isco new avam x28

### PR DESCRIPTION
**Pull request by hamidsoleymani**
AvamNew mapped to x28 code

### CODE REVIEW

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchOccupationLabelIndexer.java [hunk 1]
```diff
@@ -108,7 +106,8 @@ public void reindexOccupationLabel() {
                     .orElse(null);
 
             case X28:
-                return occupationLabelMappingRepository.findOneByX28Code(professionCode)
+                final Optional<OccupationLabelMapping> oneByX28Code = occupationLabelMappingRepository.findOneByX28Code(professionCode);
```
Comment (birom) [inline reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchOccupationLabelIndexer.java]: IMHO, this does not make the code more readable. It makes the code inconsistent. Even in the same method there are 2 different solutions.

Comment (hamidsoleymani) [reply reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchOccupationLabelIndexer.java]: reverted

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml [hunk 2]
```diff
@@ -84,4 +84,12 @@
                   tableName="reporting_obligation"/>
     </changeSet>
 
+    <changeSet id="20191114180000-1" author="haso" runOnChange="true">
+        <validCheckSum>ANY</validCheckSum>
+        <delete tableName="occupation_label_mapping_x28"></delete>
```
Comment (fabiangotzen) [inline reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: I think we shouldn't delete this. Only add the additional x28 mappings.

Comment (hamidsoleymani) [reply reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: new row will be added without deleting old rows. 
constraint unique=true on column x28_code is now changed to false

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml [hunk 3]
```diff
@@ -62,26 +70,62 @@
...
-    </changeSet>
 
     <changeSet id="20191023180000-2" author="haso" runOnChange="true">
         <validCheckSum>ANY</validCheckSum>
         <loadData file="config/liquibase/changelog/data/chisco/out/reporting-obligation.csv"
                   tableName="reporting_obligation"/>
     </changeSet>
 
+    <!-- 1. add new column without constraints-->
+    <changeSet id="2019-11-19_1" author="haso">
```
Comment (birom) [inline reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: Please respect the current naming convention. Example changesetId:
```
 <changeSet id="20191008180000-5" author="haso" runOnChange="true">
        <validCheckSum>ANY</validCheckSum>
....
```

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml [hunk 4]
```diff
@@ -62,26 +70,62 @@
...
+        <addColumn tableName="occupation_label_mapping_x28">
+            <column name="x28_code_new" type="varchar(10)">
+                <constraints unique="false"/>
+            </column>
+        </addColumn>
+    </changeSet>
+
+    <!-- 2. copy from x28_code to x28_code_new -->
+    <changeSet id="2019-11-19_2" author="haso" runOnChange="true">
```
Comment (birom) [inline reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: Please respect the current naming convention for the changesetId.

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml [hunk 5]
```diff
@@ -62,26 +70,62 @@
...
+        <sql>UPDATE occupation_label_mapping_x28
+            SET x28_code_new = x28_code;
+        </sql>
+    </changeSet>
+
+    <!-- 3. delete column x28_code   -->
+    <changeSet id="2019-11-19_3" author="haso">
```
Comment (birom) [inline reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: Please respect the current naming convention for the changesetId.

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml [hunk 6]
```diff
@@ -62,26 +70,62 @@
...
+        <dropColumn columnName="x28_code"
+                    schemaName="public"
+                    tableName="occupation_label_mapping_x28">
+            <!-- <column name="x28_code" type="varchar(10)"/>-->
+        </dropColumn>
+    </changeSet>
+
+    <!-- 4. rename column x28_code_new to x28_code  -->
+    <changeSet id="2019-11-19_4" author="haso">
```
Comment (birom) [inline reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: Please respect the current naming convention for the changesetId.

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml [hunk 7]
```diff
@@ -62,26 +70,62 @@
...
+        <renameColumn
+            newColumnName="x28_code"
+            oldColumnName="x28_code_new"
+            schemaName="public"
+            tableName="occupation_label_mapping_x28"/>
+    </changeSet>
+
+    <!-- 5. load data into table  -->
+    <changeSet id="2019-11-19_5" author="haso" runOnChange="true">
```
Comment (birom) [inline reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: Please respect the current naming convention for the changesetId.

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ReportingObligationServiceImpl.java [hunk 8]
```diff
@@ -43,11 +43,8 @@ public ReportingObligationServiceImpl(ReportingObligationRepository reportingObl
 
     @Override
     @Transactional(readOnly = true)
-    public Optional<ReportingObligationDTO> findReportingObligation(
-        ProfessionCodeDTO professionCode,
-        String cantonCode) {
-        return resolveProfessionCode(professionCode)
-            .map(professionCodeDTO -> checkReportingObligation(cantonCode, professionCodeDTO));
+    public Optional<ReportingObligationDTO> findReportingObligation(ProfessionCodeDTO professionCode, String cantonCode) {
+        return resolveProfessionCode(professionCode).map(professionCodeDTO -> checkReportingObligation(cantonCode, professionCodeDTO));
```
Comment (birom) [inline reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ReportingObligationServiceImpl.java]: Please do not change this line .The formatting was intentional. Thx.

File: reference-app/src/main/java/ch/admin/seco/service/reference/web/rest/ReportingObligationResource.java [hunk 9]
```diff
@@ -35,17 +35,15 @@ public ReportingObligationResource(ReportingObligationService reportingObligatio
     @Timed
     public ResponseEntity<ReportingObligationDTO> hasReportingObligation(ProfessionCodeDTO professionCodeDTO) {
         log.debug("REST request to check reporting obligation for Switzerland by {}", professionCodeDTO);
-        return ResponseUtil.wrapOrNotFound(this.reportingObligationService
-            .findReportingObligation(professionCodeDTO), createCacheHeader());
+        return ResponseUtil.wrapOrNotFound(this.reportingObligationService.findReportingObligation(professionCodeDTO), createCacheHeader());
```
Comment (birom) [inline reference-app/src/main/java/ch/admin/seco/service/reference/web/rest/ReportingObligationResource.java]: Please do not change this line .The formatting was intentional. Thx.

File: reference-app/src/main/java/ch/admin/seco/service/reference/web/rest/ReportingObligationResource.java [hunk 10]
```diff
@@ -35,17 +35,15 @@ public ReportingObligationResource(ReportingObligationService reportingObligatio
...
 
     @GetMapping("/reporting-obligations/check-by-canton/{codeType}/{code}")
     @Timed
     public ResponseEntity<ReportingObligationDTO> hasReportingObligationForCanton(ProfessionCodeDTO professionCodeDTO,
-        @RequestParam(required = false) String cantonCode) {
+                                                                                  @RequestParam(required = false) String cantonCode) {
         log.debug("REST request to check reporting obligation for canton '{}' by {}", cantonCode, professionCodeDTO);
-        return ResponseUtil.wrapOrNotFound(this.reportingObligationService
-            .findReportingObligation(professionCodeDTO, cantonCode), createCacheHeader());
+        return ResponseUtil.wrapOrNotFound(this.reportingObligationService.findReportingObligation(professionCodeDTO, cantonCode), createCacheHeader());
```
Comment (birom) [inline reference-app/src/main/java/ch/admin/seco/service/reference/web/rest/ReportingObligationResource.java]: Please do not change this line .The formatting was intentional. Thx.

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchOccupationLabelIndexer.java [hunk 11]
```diff
@@ -33,13 +33,7 @@
...
 import java.util.stream.Stream;
 
-import static ch.admin.seco.service.reference.domain.enums.ProfessionCodeType.AVAM;
-import static ch.admin.seco.service.reference.domain.enums.ProfessionCodeType.BFS;
-import static ch.admin.seco.service.reference.domain.enums.ProfessionCodeType.CHISCO3;
-import static ch.admin.seco.service.reference.domain.enums.ProfessionCodeType.CHISCO5;
-import static ch.admin.seco.service.reference.domain.enums.ProfessionCodeType.SBN3;
-import static ch.admin.seco.service.reference.domain.enums.ProfessionCodeType.SBN5;
-import static ch.admin.seco.service.reference.domain.enums.ProfessionCodeType.X28;
+import static ch.admin.seco.service.reference.domain.enums.ProfessionCodeType.*;
```
Comment (birom) [inline reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchOccupationLabelIndexer.java]: Please do not change this. We do not prefer wildcard imports. Thx

File: reference-app/src/main/java/ch/admin/seco/service/reference/web/rest/ReportingObligationResource.java (hunk 10 above)
Comment (hamidsoleymani) [reply reference-app/src/main/java/ch/admin/seco/service/reference/web/rest/ReportingObligationResource.java]: ok

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchOccupationLabelIndexer.java (hunk 11 above)
Comment (hamidsoleymani) [reply reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ElasticsearchOccupationLabelIndexer.java]: ok

File: reference-app/src/main/java/ch/admin/seco/service/reference/web/rest/ReportingObligationResource.java (hunk 9 above)
Comment (hamidsoleymani) [reply reference-app/src/main/java/ch/admin/seco/service/reference/web/rest/ReportingObligationResource.java]: ok

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ReportingObligationServiceImpl.java (hunk 8 above)
Comment (hamidsoleymani) [reply reference-app/src/main/java/ch/admin/seco/service/reference/service/impl/ReportingObligationServiceImpl.java]: ok

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml (hunk 7 above)
Comment (hamidsoleymani) [reply reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: ok

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml (hunk 6 above)
Comment (hamidsoleymani) [reply reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: ok

File: reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml (hunk 5 above)
Comment (hamidsoleymani) [reply reference-app/src/main/resources/config/liquibase/changelog/20191008180000_add_ch_isco_fields.xml]: ok

**Pull request merged**